Forta-Bots-Challenge-3/src/agent.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  Finding,
+  HandleBlock,
+  getEthersProvider,
+  BlockEvent,
+  getAlerts,
+} from "forta-agent";
+import {
+  DAI_ADDRESS,
+  L1_ESCROW_ARBITRUM,
+  L1_ESCROW_FUNCTION_SIGNATURE,
+  L1_ESCROW_OPTIMISM,
+  BOT_ID,
+} from "./constants";
+import { ethers, Contract } from "ethers";
+import { getL1Finding, checkBlock } from "./utils";
+
+export function provideHandleBlock(
+  provider: ethers.providers.Provider,
+): HandleBlock {
+  return async function handleBlock(block: BlockEvent): Promise<Finding[]> {
+    const findings: Finding[] = [];
+    const { chainId } = await provider.getNetwork();
+
+    try {
+      if (chainId == 1) {
+        const daiContract = new Contract(
+          DAI_ADDRESS,
+          L1_ESCROW_FUNCTION_SIGNATURE,
+          provider,
+        );
+        findings.push(
+          await getL1Finding(
+            daiContract,
+            block.blockNumber,
+            L1_ESCROW_ARBITRUM,
+            L1_ESCROW_OPTIMISM,
+          ),
+        );
+      } else {
+        //Listen to alerts
+        const alerts = await getAlerts({
+          botIds: [BOT_ID],
+          alertId: "L1 escrow supply alert",
```

Comment:
The alertId here is not exactly the one of the L1 alert (By the way, we typically use alertIds of this form: "L1-BLOCK-CHECK-ESCROWS")